test/net/sourceforge/kolmafia/persistence/ItemFinderTest.java

```diff
@@ -0,0 +1,833 @@
+  // The string can specify a single item or a comma seperated list of items.
+  //
+  // The user can specify a "count" for each item, including wildcards:
+  // "*" means "all available and "-N" means "all except N"
+
+  // Primitives:
+  //
+  // List<String> getMatchingNames(String)
+  // String getFirstMatchingItemName(List, String, Match)
+  // AdventureResult getFirstMatchingItem(String, boolean, List, Match)
+  // AdventureResult[] getMatchingItemList(String, boolean, List, Match)
+
+  // getMatchingItemList uses getFirstMatchingItem
+  // getFirstMatchingItem uses getFirstMatchingItemName
+  // getFirstMatchingItemName uses getMatchingNames
+  // getMatchingNames simply calls ItemDatabase.getMatchingNames
+
+  // The first thing to test is a string containing a single item
+  //
+  // "" -> ERROR
+  // "bogus" -> ERROR
+  // seal tooth
+  // 1 Meat
+  // mmj -> magical mystery juice
+  // 1337 7r0uZ0RZ
+  // Tea, Earl Grey, Hot
+  // [7961]Staff of Ed
+  // [7961]
+  //  ¶7961
+  //
+  // The "pilcrow" character followed by itemID was invented by Jason
+  // Harper to allow ASH RuntimeLibrary methods to call CLI commands
+  // ("buy", "closet", etc.) with no ambiguity.
+  //
+  // The [itemId] forms were added later as general disambiguation for
+  // multiple object types in ASH
+  //
+  // Item names can contain a comma. Even though the parameter string can
+  // contain multiple comma-separated items, we do not allow or expect you to
+  // escape the comma. Instead, you can only specify such an item (by name) if
+  // it is the entire parameter string.
+  //
+  // Although the user can specify N of an item, item names can start with a
+  // number. Tricky!
+  //
+  // Lastly notice that we do "fuzzy matching" using ItemDatabase.getMatchingNames(),
+  // which uses StringUtilities.getMatchingNames().
+  // That package deserves its own test suite.
+  //
+  // also:
+  //
+  // 3215 Meat -> MEAT, not an item
+
+  // *********************** getFirstMatchingItem ***********************
+  //
+  // That method assumes its parameter is the name of a single item,
+  // possible with a count. Assuming it unambiguously names a single
+  // item, it returns an AdventureResult with itemId and count.
+
+  @Test
+  public void itShouldParseSingleItem() {
+
+    // This set of tests will use null as the source list. That means
+    // that it will not restrict its matches to what is available in a
+    // specific list. The only exception is if you use wildcard counts -
+    // * or -N - in which case, it assumes inventory. We'll test wildcard
+    // counts separately.
+
+    AdventureResult item;
+    String message;
+    String parameter;
+
+    // Test successful exact match for entire string.
+    item = ItemFinder.getFirstMatchingItem("toilet paper", false, null, Match.ANY);
+    assertEquals(StaticEntity.getContinuationState(), MafiaState.CONTINUE);
+    assertTrue(item != null);
+    assertTrue(item.getItemId() == ItemPool.TOILET_PAPER);
```

Comment:
`assertEquals(ItemPool.TOILET_PAPER, item.getItemId())` should give a slightly better message in the case of failure. I think it's better to default to using `assertEquals(x, y)` over `assertTrue(x == y)`.

Reply:
Yes, this makes sense. Does assertEquals actually use the equals() method? (One would hope)
I'll experiment and convert, as appropriate.

Reply:
All done.